extension/tensor/tensor_ptr.h

```diff
@@ -79,7 +83,8 @@ class TensorPtr : private std::unique_ptr<exec_aten::Tensor> {
 
  private:
   friend TensorPtr make_tensor_ptr(const TensorPtr& tensor);
-  TensorImplPtr tensor_impl;
+  mutable exec_aten::Tensor tensor_{nullptr};
+  TensorImplPtr tensor_impl_;
```

Comment:
note that `tensor_` and `tensor_impl_` are redundant -- under the hood, they are both just `TensorImpl*` that point to the same thing. PyTorch has also wrestled with this problem. IIRC I thought I finally cracked it within the past year or so, but never committed the PR because I didn't have a clear reason; I will try to dig it up.

Reply:
I was thinking of https://github.com/pytorch/pytorch/pull/95418 . We don't have the same problem here; rather than grafting not-reference-counting onto a reference-counting Tensor, we want to graft reference counting onto a not-reference-counting Tensor. I'll give it some more thought.

Reply:
this is solvable if we are able to remove get() and rely on -Waddress-of-temporary. diffs coming.